### DISCUSSION TITLE
Anyone did the migration manually?

### DISCUSSION THREAD

**Discussion by lkj4** [sveltejs/kit#6189] [Q&A]
I've a medium-size project and I'm wondering if I just should do the changes manually instead of running the svelte-migration script. However, running the script shouldn't mess anything up and if, I can still `git restore .` and this other git command which also removes newly added files...

**Comment by nikfp** [comment]
FWIW, the couple of projects I've migrated with the script both went well. I'd recommend that you do a git commit before hand, then run through the instructions [in this discussion](https://github.com/sveltejs/kit/discussions/5774), and if anything goes wrong you can `git reset --hard HEAD` and you are back at your starting point again.

**Comment by jesseskinner** [comment]
The migration tool is really good, I would use it on any sized project. It does as much as it can on its own, and then forces you to walk through anything that may need special attention, which is great, way lowering your chance of breaking something.

**Comment by danawoodman** [comment]
I don't see any compelling reason not to use it on any project. Doing it manually seems a waste of time and energy.

**Comment by ObieMunoz** [comment]
Use the tool.

**Comment by DrDanRyan** [comment]
On one of my projects the script crashed halfway through my routes and I did the remaining half without the aid of the tool. It actually wasn't too bad after learning the pattern of what you have to do. This was my 2nd project that I migrated though, so going through it once already really helped me understand everything that needed to change.

**Comment by janzheng** [comment]
The tool will regard your non-endpoint `[helpers].js` files as server endpoints and turn them into `/folder/+server.js` which is super annoying... other than, use the tool!

**Comment by janzheng** [comment]
One thing I wish the tool did was instead of add `@migration task: Add data prop` that it would also do that automatically, wrapping any variables into data. I'm finding myself doing this a lot even with the tool. e.g.

```
export let data
export let {blogposts} =  data
```

**Reply by danawoodman** [reply]
I don't think you want to export `{ blogposts }` but instead do assignment?

```ts
export let data

$: blogposts = data.blogposts

// or

$: ({ blogposts} = data)
```
